src/coreclr/vm/jitinterface.cpp

```diff
@@ -11725,16 +11725,24 @@ void CEEJitInfo::recordRelocation(void * location,
                 }
                 else
                 {
-                    //
-                    // When m_fAllowRel32 == FALSE, the JIT will use a REL32s for direct code targets only.
-                    // Use jump stub.
-                    //
-                    delta = rel32UsingJumpStub(fixupLocation, (PCODE)target, m_pMethodBeingCompiled, NULL, false /* throwOnOutOfMemoryWithinRange */);
-                    if (delta == 0)
+                    if (m_fJumpStubOverflow)
                     {
-                        // This forces the JIT to retry the method, which allows us to reserve more space for jump stubs and have a higher chance that
-                        // we will find space for them.
-                        m_fJumpStubOverflow = TRUE;
+                        // Do not attempt to allocate more jump stubs. We are going to throw away the code and retry anyway.
+                        delta = 0;
```

Comment:
Is my understanding of what happens in the test correct?

1. We have a first JIT attempt that sets `m_fAllowRel32 = false`
2. The second JIT attempt fails to allocate a jump stub and sets `m_fJumpStubOverflow = false`
3. The third try allocates a new code heap without any address range constraints and puts everything in there

Reply:
Or is 1 and 2 happening during the same attempt? And the slowness of the compilation was because of needing to find 284k jump stubs, which all go into the `ReserveWithinRange` path that can be quite slow?